### ISSUE TITLE
[rllib] Why can't I get a high reward when training PPO in BreakoutNoFrameskip-v4？

### ISSUE THREAD

**Issue by iiapache** [ray-project/ray#16693]
I trained PPO in env BreakoutNoFrameskip-v4 using the almost same config as: [https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/ppo/atari-ppo.yaml](https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/ppo/atari-ppo.yaml). However, I can not get a high episode_mead_reward（around 2.0）, which should around 200.
     
![image](https://user-images.githubusercontent.com/4160477/123573965-93225600-d801-11eb-9742-8d2f459a4682.png)

The main package Version I used:
    python: 3.6.8
    ray: 1.2.0
    tensorflow: 2.2.0
    gym: 0.18.0

The config in https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/ppo/atari-ppo.yaml:

> `# Runs on a single g3.16xl node
> # See https://github.com/ray-project/rl-experiments for results
> atari-ppo:
>     env:
>         grid_search:
>             - BreakoutNoFrameskip-v4
>             - BeamRiderNoFrameskip-v4
>             - QbertNoFrameskip-v4
>             - SpaceInvadersNoFrameskip-v4
>     run: PPO
>     config:
>         # Works for both torch and tf.
>         framework: tf
>         lambda: 0.95
>         kl_coeff: 0.5
>         clip_rewards: True
>         clip_param: 0.1
>         vf_clip_param: 10.0
>         entropy_coeff: 0.01
>         train_batch_size: 5000
>         rollout_fragment_length: 100
>         sgd_minibatch_size: 500
>         num_sgd_iter: 10
>         num_workers: 10
>         num_envs_per_worker: 5
>         batch_mode: truncate_episodes
>         observation_filter: NoFilter
>         model:
>             vf_share_layers: true
>         num_gpus: 1`

The config I used：

>        `{
>   "_fake_gpus": false,
>   "_use_trajectory_view_api": true,
>   "batch_mode": "truncate_episodes",
>   "callbacks": "<class 'ray.rllib.agents.callbacks.DefaultCallbacks'>",
>   "clip_actions": true,
>   "clip_param": 0.1,
>   "clip_rewards": true,
>   "collect_metrics_timeout": 180,
>   "compress_observations": false,
>   "create_env_on_driver": false,
>   "custom_eval_function": null,
>   "custom_resources_per_worker": {},
>   "eager_tracing": false,
>   "entropy_coeff": 0.01,
>   "entropy_coeff_schedule": null,
>   "env": "BreakoutNoFrameskip-v4",
>   "env_config": {},
>   "evaluation_config": {},
>   "evaluation_interval": 1,
>   "evaluation_num_episodes": 10,
>   "evaluation_num_workers": 0,
>   "exploration_config": {
>     "type": "StochasticSampling"
>   },
>   "explore": true,
>   "extra_python_environs_for_driver": {},
>   "extra_python_environs_for_worker": {},
>   "fake_sampler": false,
>   "framework": "torch",
>   "gamma": 0.99,
>   "grad_clip": null,
>   "horizon": null,
>   "ignore_worker_failures": false,
>   "in_evaluation": false,
>   "input": "sampler",
>   "input_evaluation": [
>     "is",
>     "wis"
>   ],
>   "kl_coeff": 0.5,
>   "kl_target": 0.01,
>   "lambda": 0.95,
>   "local_tf_session_args": {
>     "inter_op_parallelism_threads": 8,
>     "intra_op_parallelism_threads": 8
>   },
>   "log_level": "DEBUG",
>   "log_sys_usage": true,
>   "logger_config": null,
>   "lr": 5e-05,
>   "lr_schedule": null,
>   "memory": 0,
>   "memory_per_worker": 0,
>   "metrics_smoothing_episodes": 100,
>   "min_iter_time_s": 0,
>   "model": {
>     "conv_activation": "relu",
>     "conv_filters": null,
>     "custom_action_dist": null,
>     "custom_model": null,
>     "custom_preprocessor": null,
>     "dim": 84,
>     "fcnet_activation": "tanh",
>     "fcnet_hiddens": [
>       256,
>       256
>     ],
>     "framestack": true,
>     "free_log_std": false,
>     "grayscale": false,
>     "lstm_cell_size": 256,
>     "lstm_use_prev_action": false,
>     "lstm_use_prev_action_reward": false,
>     "lstm_use_prev_reward": false,
>     "max_seq_len": 20,
>     "no_final_linear": false,
>     "use_lstm": false,
>     "vf_share_layers": true,
>     "zero_mean": true
>   },
>   "monitor": false,
>   "multiagent": {
>     "count_steps_by": "env_steps",
>     "observation_fn": null,
>     "policies": {},
>     "policies_to_train": null,
>     "policy_mapping_fn": null,
>     "replay_mode": "independent"
>   },
>   "no_done_at_end": false,
>   "normalize_actions": false,
>   "num_cpus_for_driver": 1,
>   "num_cpus_per_worker": 1,
>   "num_envs_per_worker": 5,
>   "num_gpus": 0,
>   "num_gpus_per_worker": 0,
>   "num_sgd_iter": 10,
>   "num_workers": 10,
>   "object_store_memory": 0,
>   "object_store_memory_per_worker": 0,
>   "observation_filter": "NoFilter",
>   "optimizer": {},
>   "output": null,
>   "output_compress_columns": [
>     "obs",
>     "new_obs"
>   ],
>   "output_max_file_size": 67108864,
>   "postprocess_inputs": false,
>   "preprocessor_pref": "deepmind",
>   "remote_env_batch_wait_ms": 0,
>   "remote_worker_envs": false,
>   "replay_sequence_length": 1,
>   "rollout_fragment_length": 100,
>   "sample_async": false,
>   "sample_collector": "<class 'ray.rllib.evaluation.collectors.simple_list_collector.SimpleListCollector'>",
>   "seed": null,
>   "sgd_minibatch_size": 500,
>   "shuffle_buffer_size": 0,
>   "shuffle_sequences": true,
>   "simple_optimizer": true,
>   "soft_horizon": false,
>   "synchronize_filters": true,
>   "tf_session_args": {
>     "allow_soft_placement": true,
>     "device_count": {
>       "CPU": 1
>     },
>     "gpu_options": {
>       "allow_growth": true
>     },
>     "inter_op_parallelism_threads": 2,
>     "intra_op_parallelism_threads": 2,
>     "log_device_placement": false
>   },
>   "timesteps_per_iteration": 0,
>   "train_batch_size": 5000,
>   "use_critic": true,
>   "use_gae": true,
>   "vf_clip_param": 10.0,
>   "vf_loss_coeff": 1.0,
>   "vf_share_layers": true
> }`

      

  Appreciate that anyone can help to figure out the problem!

**Comment by iiapache** [comment]
Can anyone help solve this problem?

**Comment by richardliaw** [comment]
Try taking a look at this? https://github.com/ray-project/rl-experiments/tree/master/atari-ppo/2020-01-21/PPO_BreakoutNoFrameskip-v4_0_use_pytorch%3DFalse_2020-01-21_10-10-abcdefg

**Comment by iiapache** [comment]
> Try taking a look at this? https://github.com/ray-project/rl-experiments/tree/master/atari-ppo/2020-01-21/PPO_BreakoutNoFrameskip-v4_0_use_pytorch%3DFalse_2020-01-21_10-10-abcdefg

Thanks for reply！I have followed the config, however, the result is the same. Although 13M timesteps have been trained, the reward still remain about 2.  It's really strange.
